---
title: "evidence2"
output:
  pdf_document: default
  html_document: default
date: "2025-05-02"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

Load all of the libraries that we'll be using for this project

```{r echo=FALSE}
# BioCManager library installation
#if (!requireNamespace("BiocManager", quietly = TRUE)){
#  install.packages("BiocManager")
#  BiocManager::install("msa")
#  BiocManager::install("DECIPHER")
#  BiocManager::install("phangorn")
#  BiocManager::install("treeio")
#}

# Library loading
library(ade4)
library(ape)
library(adegenet)
library(Biostrings)
library(DECIPHER)
library(ggplot2)
library(reshape2)
library(dplyr)
library(stringr)
library(tidyr)
library(phangorn)
library(ggplot2)
library(ggthemes)
library(ggrepel)
library(msa)
library(ggtree)
library(treeio)
```

Performing sequence loading

```{r}
# List of countries where COVID killed the most people
countries <- c("usa", "china", "india", "france", "germany", "brazil", "south_korea", "japan", "italy", "uk")

# Map the names to DNA String Sets
sequences <- sapply(countries, function(x) readDNAStringSet(paste("./sequences/", x, ".fasta", sep="")))

# Convert to DNABin types
dna_bins <- do.call(c, sapply(sequences, as.DNAbin))

# Labeling and Display
country_titles <- sapply(countries, function(string) str_to_title(str_replace(string, "_", " ")))

# Create a  Dataframe from the Sequences
sequence_frame <- data.frame(
  country_key = countries,
  country_title  = country_titles,
  sequence_length = sapply(sequences, width),
  a_frequency = sapply(sequences, function(seq) letterFrequency(seq, 'A')),
  c_frequency = sapply(sequences, function(seq) letterFrequency(seq, 'C')),
  g_frequency = sapply(sequences, function(seq) letterFrequency(seq, 'G')),
  t_frequency = sapply(sequences, function(seq) letterFrequency(seq, 'T'))
)

# Print Genome Sequence Lengths
print("The sequence lengths for each genome are as follows")
sequence_frame[3]

# Print Genome Sequence Nitrogenous Base Frequencies
print("The frequencies for each base in each genome are as follows:")
sequence_frame[c(4, 5, 6, 7)]

```

Convert to data frame and graph

```{r}
# Plot Frequencies
ggplot(
  data = sequence_frame %>% gather(Frequencies, Frequency, -country_title, -country_key, -sequence_length),
  aes(x = country_title, y = Frequency, fill = Frequencies)
) +
  geom_bar(stat = 'identity', position = 'dodge') +
  xlab("Countries") +
  scale_fill_discrete(
    labels = c(
      "a_frequency" = "Number of A Bases",
      "c_frequency" = "Number of C Bases",
      "g_frequency" = "Number of G Bases",
      "t_frequency" = "Number of T Bases"
  ))

```

```{r}
# Read FASTAs Again but in a Different Format (idealy, this shouldn't be the case but we couldn't find how to do it)
countries <- c("usa", "china", "india", "france", "germany", "brazil", "south_korea", "japan", "italy", "uk")

# Place Sequences into a DNAStringSet
all_sequences <- DNAStringSet()
for (country in countries) {
  seq <- readDNAStringSet(paste0("./sequences/", country, ".fasta"))
  names(seq) <- country  # Explicitly name each sequence
  all_sequences <- c(all_sequences, seq)
}

# Load Mortality Rates
# 3. Prepare mortality data with EXACT MATCH to tip labels
mortality_data <- data.frame(
  label = countries,  # Must match tree$tip.label
  Cases = c(103436829, 99380363, 45043415, 39016278, 38437756, 37511921, 
            34571873, 33803572, 26826486, 24992089),
  Deaths = c(1201488, 122358, 533641, 168091, 174979, 702116, 
             35934, 74694, 197542, 232112)
  
) %>% 
  mutate(
    Mortality = round(Deaths/Cases * 100, 2),
    Country = stringr::str_to_title(gsub("_", " ", label))
  )

mortality_data$Rates = mortality_data$Deaths / mortality_data$Cases

# Align DNA Sequences
aligned <- AlignSeqs(all_sequences)
dna <- as.matrix(as.DNAbin(aligned))

# Stop if Our Sequences are not Aligned
stopifnot(length(unique(ncol(dna))) == 1)

# DNA Distribution
dna_distro <- dist.dna(dna, model = "TN93")

# Create NJ Tree From DNA Distro
phylotree <- nj(dna_distro)



# Calculate Bootstrapping Values
boots <- boot.phylo(
  phylotree, 
  dna, 
  function(e) root(nj(dist.dna(e, model = "TN93")), 1),
  B = 100,
  quiet = TRUE
)

# Plot our Phylo Tree with Bootstrap Values
myPal <- colorRampPalette(c("red","yellow","green","blue"))
mortalityPalette <- colorRampPalette(c("red", "blue"))
plot(phylotree, cex = 0.6, main = "NJ Tree")

```

```{r}
# Convert our DNA Distribution and Plot it in a Table
adj_matrix <- as.matrix(dna_distro)
table.paint(
  adj_matrix,
  cleg = 0,
  clabel.row = 0.25,
  clabel.col = 0.25
)
```

```{r}
p <- ggtree(phylotree) %<+% mortality_data +
  geom_tiplab(aes(label = paste0(Country, "\n", Mortality, "%")), 
             size = 3, hjust = -0.1) +
  geom_tippoint(aes(color = Mortality), size = 5) +
  scale_color_gradient(low = "blue", high = "red", na.value = "gray") +
  xlim(0, max(dna_distro) * 1.3) +
  labs(title = "Phylogenetic Tree with Mortality Rates",
      color = "Mortality (%)") +
  theme_tree2()

print(p)
```